### ISSUE TITLE
Invalid state after robust reconnect

### ISSUE THREAD

**Issue by tilsche** [metricq/metricq#23]
Sometimes things go wrong when reconnecting with the robust channels.

Case 1, manager:

```
Mar 29 10:06:20 igel metricq.agent[23517]: closing connection to (320, "CONNECTION_FORCED - broker forced connection closure with reason 'shutdown'")
Mar 29 10:06:20 igel metricq.agent[23517]: exception in event loop: Task exception was never retrieved
Mar 29 10:06:20 igel metricq.agent[23517]: Future: <Task finished coro=<_wrap_awaitable() done, defined at /home/service/envs/manager/lib/python3.7/asyncio/tasks.py:596> exception=<Connectio
Mar 29 10:06:20 igel metricq.agent[23517]: Exception: (320, "CONNECTION_FORCED - broker forced connection closure with reason 'shutdown'") (ConnectionClosed)
Mar 29 10:06:20 igel metricq.agent[23517]: closing connection to (320, "CONNECTION_FORCED - broker forced connection closure with reason 'shutdown'")
Mar 29 10:06:20 igel metricq.agent[23517]: exception in event loop: Task exception was never retrieved
Mar 29 10:06:20 igel metricq.agent[23517]: Future: <Task finished coro=<_wrap_awaitable() done, defined at /home/service/envs/manager/lib/python3.7/asyncio/tasks.py:596> exception=<Connectio
Mar 29 10:06:20 igel metricq.agent[23517]: Exception: (320, "CONNECTION_FORCED - broker forced connection closure with reason 'shutdown'") (ConnectionClosed)
```
Sometimes it works after a while - maybe management channel reconnected before data channel but then an RPC wanted to bind some things in the data channel. Sometimes only the restart helps.

Case grafana:

```
Mar 29 10:28:36 igel metricq-grafana[21324]: aiormq.exceptions.ChannelNotFoundEntity: NOT_FOUND - no queue 'history-history-grafana' in vhost '/'
Mar 29 10:28:36 igel metricq-grafana[21324]:     raise self.exception from e
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/site-packages/aiormq/base.py", line 24, in __await__
Mar 29 10:28:36 igel metricq-grafana[21324]:     return (yield from awaitable.__await__())
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/asyncio/tasks.py", line 603, in _wrap_awaitable
Mar 29 10:28:36 igel metricq-grafana[21324]:     raise self.exception from e
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/site-packages/aiormq/base.py", line 24, in __await__
Mar 29 10:28:36 igel metricq-grafana[21324]:     return await self.create_task(func(self, *args, **kwargs))
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/site-packages/aiormq/base.py", line 161, in wrap
Mar 29 10:28:36 igel metricq-grafana[21324]:     arguments=arguments,
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/site-packages/aiormq/channel.py", line 505, in queue_declare
Mar 29 10:28:36 igel metricq-grafana[21324]:     return await fut
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/asyncio/tasks.py", line 388, in wait_for
Mar 29 10:28:36 igel metricq-grafana[21324]:     ), timeout=timeout, loop=self.loop
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/site-packages/aio_pika/queue.py", line 86, in declare
Mar 29 10:28:36 igel metricq-grafana[21324]:     await self.declare()
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/site-packages/aio_pika/robust_queue.py", line 50, in on_reconnect
Mar 29 10:28:36 igel metricq-grafana[21324]:     await queue.on_reconnect(self)
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/site-packages/aio_pika/robust_channel.py", line 62, in on_reconnect
Mar 29 10:28:36 igel metricq-grafana[21324]:     await channel.on_reconnect(self, number)
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/site-packages/aio_pika/robust_connection.py", line 104, in _on_reconnect
Mar 29 10:28:36 igel metricq-grafana[21324]: Traceback (most recent call last):
Mar 29 10:28:36 igel metricq-grafana[21324]: The above exception was the direct cause of the following exception:
Mar 29 10:28:36 igel metricq-grafana[21324]: concurrent.futures._base.CancelledError
Mar 29 10:28:36 igel metricq-grafana[21324]:     return (yield from self.task.__await__(*args, **kwargs))
Mar 29 10:28:36 igel metricq-grafana[21324]:   File "/home/service/envs/grafana/lib/python3.7/site-packages/aiormq/base.py", line 22, in __await__
```

Seems like a reconnect gets confused trying to passively declare the exclusive RPC response queues.

**Comment by bmario** [comment]
Should be fixed as well in v1.1.0?

**Comment by phijor** [comment]
Closed in #46.
